src/Http/CurlClient.php

```diff
@@ -11,10 +11,21 @@
  */
 class CurlClient implements ClientInterface
 {
+    /**
+     * Override this method if you need to set any special parameters like disable CURLOPT_SSL_VERIFYHOST and CURLOPT_SSL_VERIFYPEER.
+     * @return false|resource
+     */
+    protected function createCurl()
```

Comment:
np: naming of `createCurl()` vs calling then `curl_init()`. The function suggests to initialize curl, the function to create it. Maybe `initCurl()` would be more straight forward, but again no hard feelings just some thoughts.